class-7/aws_s3_tool/lib/aws_s3_tool.ex

```diff
@@ -0,0 +1,45 @@
+defmodule AwsS3Tool do
+  @moduledoc """
+  Documentation for `AwsS3Tool`.
+  """
+
+  @doc """
+  Uploads any file to an Aws S3 Bucket.
+
+  ## Examples
+      iex> AwsS3Tool.upload("Users/owner/Documents/file_to_upload")
+  """
+  def upload(file_path) do
+    case read_file(file_path) do
+      {:ok, binary} ->
+        s3_upload(binary, file_path)
+
+      {:error, _reason} ->
+        "It is not posible to read file in: #{file_path}"
+    end
+  end
```

Comment:
```suggestion
  def upload(file_path) do
    with binary <- read_file(file_path) do
         s3_upload(binary, file_path)
    end
  end
```

Te recomiendo estas suggestions, cuentame que te parece.

Reply:
Y el mensaje de error a dónde quedaría en este caso?

Reply:
@lucdelrio queda en el sistema, OTP internamente va a reinciar el proceso, esto son temas avanzados, pero te recomiendo leer!! La manera en que lo trabajaste ES CORRECTA, pero solamente quiero que practiques de la otra forma tambien

https://elixir-lang.org/getting-started/try-catch-and-rescue.html

Reply:
Gracias Alberto! Queda más claro ahora. Es algo propio del manejo del módulo File también.